ogr/services/github/service.py

```diff
@@ -178,3 +179,41 @@ def project_create(
     def get_pygithub_instance(self, namespace: str, repo: str) -> PyGithubInstance:
         token = self.authentication.get_token(namespace, repo)
         return PyGithubInstance(login_or_token=token)
+
+    def list_projects(
+        self,
+        namespace: str = None,
+        user: str = None,
+        search_patter: str = None,
+        language: str = None,
+    ) -> List[GitProject]:
+        g = self.github
+
+        # Get the locals for the function
+        passed_args = locals()
+
+        # Get the arguments which are not None and remove self and g
+        args_for_query = {
+            k: v
+            for k, v in passed_args.items()
+            if v and k not in ["self", "g", "search_patter"]
+        }
```

Comment:
I would try to be explicit and build the query manually. This looks very nice but it's maybe harder to read.

(And `mypy` would like the explicitness as well.)

Reply:
Like you've done in case of GitLab.